### ISSUE TITLE
b-modal - accessibility when passing header slot

### ISSUE THREAD

**Issue by jamart2005** [bootstrap-vue/bootstrap-vue#3828]
### Describe the bug
When passing modal-header slot, modal's aria-labelledby do not seem to point to modal-header.

### Steps to reproduce the bug

1. Go to https://jsfiddle.net/c3ufz2a4/
2. Click on **Show Modal**

### Expected behavior
Header should have been read when modal shows

### Versions

**Environment:**
- OS: Windows 10
- Browser: Chrome
- Version: 74
- A11y - JAWS and NVDA 

### Demo link
https://jsfiddle.net/c3ufz2a4/

**Comment by tmorehouse** [comment]
Since the header is not generated by BootstrapVue in this case (when user provides their own header), it does not know the header's ID so it cannot associate an ID it doesn't know of to the `aria-labelledby` attribute.

**Comment by tmorehouse** [comment]
But note that `aria-describedby` will be set to point to the modal  in this situation.

**Comment by tmorehouse** [comment]
See related discussion at: https://github.com/bootstrap-vue/bootstrap-vue/issues/3801

**Comment by jamart2005** [comment]
We still have the header tag which has b-modal generated id. the header slot put its contents inside of that. Don't we want to use that?

There is currently a workaround for this, which is to let users that pass header-slot to define a label by aria-label prop instead.  I was just thinking if having it point to header tag as aria-labelledby as default would be a better behavior.

**Comment by tmorehouse** [comment]
The only issue when pointing to the full header is that screen readers will also read out the "x" close button as part of the label, which doesn't always make sense.

**Comment by tmorehouse** [comment]
https://github.com/bootstrap-vue/bootstrap-vue/pull/3715

https://github.com/bootstrap-vue/bootstrap-vue/issues/3712#issuecomment-513049766

**Comment by tmorehouse** [comment]
The only other option, possibly, would be to add an `aria-labelledby` prop, which would be only used if the header slot is used, and would need to point to a user assigned ID (the ID would also have a restriction that the element that has that ID be contained within the modal)

**Comment by jamart2005** [comment]
Ah ok. makes sense when users start adding buttons to header slot and if that is the primary purpose of that then it make sense to not do it.  The workaround sounds good also! 
Thank you Troy!

**Comment by tmorehouse** [comment]
If your goal is to just hide the header close button, then using the prop `hide-header-close` is your best friend. Otherwise the `aria-label` prop is your fiend.
